.ci/packaging.groovy

```diff
@@ -209,6 +212,29 @@ def tagAndPush(name){
   }
 }
 
+def runE2ETestForPackages(){
```

Comment:
I think we are running the tests too early. Let me explain: as we are building different projects in parallel (i.e. metricbeat and filebeat) it could be possible that if the filebeat build is faster, then it reaches the e2e tests stage when the filebeat one has not created its artifacts. Therefore the e2e tests could potentially fail.

I could see it valid to run the e2e tests right after the release stage, so we ensure all artifacts are there. On the other hand, the tests won't be run until the slowest build hadn't finished.

As we discussed, it could be great to decide what to build in an earlier stage, something like _prepareBuildContext_ or similar, where we calculate what to build to be tested.

Reply:
if the step pushCIDockerImages() is finish the packages are generated and deployed, I not get your concern.

Reply:
yes, the `pushCIDockerImages` method is executed per BEATS_FOLDER, so in the context of `BEATS_FOLDER=filebeat` it will push the images for filebeat, only. And maybe we want to test the metricbeat ones too (i.e. elastic-agent starting both processes)

Reply:
this is part of the dependencies of the elastic agent, What are the conditions that define if we have to test Filebeat, Metricbeat, or both?
* If there are changes in Metricbeat Should we test the Elastic Agent? Should we test Elastic Agent+Filebeat?
* If there are changes in Filebeat Should we test the Elastic Agent? Should we test Elastic Agent+Metricbeat?
* If there are changes in Elastic Agent Should we test the Elastic Agent+Filebeat+Metricbeat?

Reply:
If those are direct dependencies then https://github.com/elastic/beats/blob/d2a89224e9718cc4e362b5aa38935465c4552df6/Jenkinsfile#L1364-L1381 could be the one to define that particular workflow for the packaging.

Otherwise, where those dependencies are defined?

Reply:
I do not think they are defined yet. I did a best effort here: https://github.com/elastic/e2e-testing/issues/309#issuecomment-697444677